ui/app/AppLayouts/Onboarding/views/ProfileFetchingView.qml

```diff
@@ -59,56 +55,167 @@ Item {
+                Text {
+                    id: loaded
+                    anchors.right: parent.right
+                    anchors.verticalCenter: parent.verticalCenter
+                    visible: !indicator.visible
+                    rightPadding: 4
+                    font.pixelSize: Constants.onboarding.profileFetching.entityProgressFontSize
+                    color: Theme.palette.baseColor1
+                    text: "%1/%2".arg(model.loadedMessages).arg(model.totalMessages)
+                }
+
+                StatusProgressBar {
+                    id: progress
+                    anchors.top: entity.bottom
+                    anchors.left: entity.left
+                    anchors.right: parent.right
+                    anchors.topMargin: 4
+                    height: 3
+                    from: 0
+                    to: model.totalMessages
+                    value: model.loadedMessages
+                    backgroundColor: Theme.palette.baseColor5
+                    backgroundBorderColor: backgroundColor
+                    fillColor: {
+                        if (model.totalMessages > 0 && model.totalMessages === model.loadedMessages)
+                            return Theme.palette.successColor1
+                        return Theme.palette.primaryColor1
+                    }
+                }
             }
+
+            ScrollBar.vertical: ScrollBar {}
         }
 
         StatusButton {
             id: button
-            visible: false
             Layout.alignment: Qt.AlignHCenter
+            Layout.preferredWidth: root.startupStore.currentStartupState.stateType === Constants.startupState.profileFetching? 80 : -1
             focus: true
-            text: Constants.onboarding.profileFetching.tryAgainText
-            onClicked: {
-                root.startupStore.doPrimaryAction()
-            }
-        }
+            enabled: root.startupStore.currentStartupState.stateType !== Constants.startupState.profileFetching
 
-        StatusBaseText {
-            id: link
-            visible: false
-            Layout.alignment: Qt.AlignHCenter
-            font.pixelSize: Constants.keycard.general.buttonFontSize
-            text: Constants.onboarding.profileFetching.createNewProfileText
-            color: Theme.palette.primaryColor1
-            font.underline: linkMouseArea.containsMouse
-            MouseArea {
-                id: linkMouseArea
-                anchors.fill: parent
-                cursorShape: Qt.PointingHandCursor
-                hoverEnabled: true
-                onClicked: {
-                    root.startupStore.doSecondaryAction()
+            Timer {
+                id: timer
+                interval: 1000
+                running: root.startupStore.currentStartupState.stateType === Constants.startupState.profileFetching
+                repeat: true
+                onTriggered: {
+                    d.counter--
+                    let timerVal = d.getCurrTimerValue()
```

Comment:
Might be safer to check if d.counter == 0. This way the logic doesn't depend on the time format.

Reply:
@alexjba thanks, I overlooked that, updated.